kv/kvserver: TestLeaseTransferReplicatesLocks failed

kv/kvserver.TestLeaseTransferReplicatesLocks [failed](https://mesolite.cluster.engflow.com/invocations/default/804f1eb1-d48d-4ffb-a567-2ee6561fe51f?testReportRun=1&testReportShard=39&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [943f82fd320a79d23fc71ddb65f0223e10713081](https://github.com/cockroachdb/cockroach/commits/943f82fd320a79d23fc71ddb65f0223e10713081):


```
=== RUN   TestLeaseTransferReplicatesLocks
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestLeaseTransferReplicatesLocks134941666
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:686: DRPC is enabled (via override by TestingGlobalDRPCOption)
    client_replica_test.go:5906: transferring to s1
    client_replica_test.go:5908: done transferring to s1
    client_replica_test.go:5926: txn1: lock acquired, waiting for txn2 cancellation
    client_replica_test.go:5939: txn2: tnx1 lock acquisition observed, starting txn
    client_replica_test.go:5965: transferring lease from s1 -> s2
    client_replica_test.go:5968: cancelling txn2
    client_replica_test.go:5928: txn1: done
    client_replica_test.go:5975: 
        	Error Trace:	pkg/kv/kvserver/client_replica_test.go:5975
        	Error:      	"0" is not greater than or equal to "1"
        	Test:       	TestLeaseTransferReplicatesLocks
    panic.go:615: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseTransferReplicatesLocks134941666
--- FAIL: TestLeaseTransferReplicatesLocks (35.04s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=39</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseTransferReplicatesLocks.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-60722

--- a/pkg/kv/kvserver/client_replica_test.go
+++ b/pkg/kv/kvserver/client_replica_test.go
@@ -5868,6 +5868,7 @@ func TestOptimisticEvalWithConcurrentWriters(t *testing.T) {
 func TestLeaseTransferReplicatesLocks(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+	skip.UnderDuress(t, "can cause uncooperative lease change under leader leases")
 
 	testutils.SetVModule(t, "cmd_lease=2")
 

